From fc8eb7eb3a76fdca2c616b8d6f3e4b90b6879f3c Mon Sep 17 00:00:00 2001
From: Sergio Pena <sergio.pena@cloudera.com>
Date: Tue, 27 Sep 2016 16:19:42 -0500
Subject: [PATCH 0772/1164] PROPOSED CDH-45249: HIVE-14848: S3 creds added to
 a hidden list by HIVE-14588 are not working on MR
 jobs

Current Hive 2.2.0-SNAPSHOT (upstream version) cannot use the new MR property "mapreduce.job.redacted-properties"
because it is available only on Hadoop 2.9.0 (not released yet).

Change-Id: Ic03dd3c737709ba3dd8ebb0dcd87358366eb284e
(cherry picked from commit 0a9471ab9fe17e76310f698c3a86e1b8179134d0)
---
 .../apache/hadoop/hive/ql/exec/mr/ExecDriver.java  |    6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecDriver.java b/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecDriver.java
index c787424..357bb74 100644
--- a/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecDriver.java
+++ b/ql/src/java/org/apache/hadoop/hive/ql/exec/mr/ExecDriver.java
@@ -170,7 +170,11 @@ public void initialize(HiveConf conf, QueryPlan queryPlan, DriverContext driverC
     initializeFiles("tmpfiles", getResource(conf, SessionState.ResourceType.FILE));
     initializeFiles("tmparchives", getResource(conf, SessionState.ResourceType.ARCHIVE));
 
-    conf.stripHiddenConfigurations(job);
+    // Hide sensitive configuration values from MR HistoryUI by telling MR to redact the following list.
+    // Note: We should not strip the values before submitting the job as there might be other variables used
+    // by MR, such as S3 credentials.
+    job.set("mapreduce.job.redacted-properties", job.get(HiveConf.ConfVars.HIVE_CONF_HIDDEN_LIST.varname));
+
     this.jobExecHelper = new HadoopJobExecHelper(job, console, this, this);
   }
 
-- 
1.7.9.5

